packages/features/bookings/lib/handleNewBooking/test/round-robin-no-hosts.test.ts

```diff
@@ -0,0 +1,238 @@
+        name: "Round Robin Host",
+        username: "round-robin-host",
+        timeZone: Timezones["+5:30"],
+        email: "round-robin-host@example.com",
+        id: 102,
+        schedules: [TestData.schedules.IstMorningShift],
+        credentials: [getGoogleCalendarCredential()],
+        selectedCalendars: [TestData.selectedCalendars.google],
+      };
+
+      const scenarioData = getScenarioData({
+        eventTypes: [
+          {
+            id: 1,
+            slotInterval: 45,
+            length: 45,
+            hosts: [
+              {
+                userId: 101,
+                isFixed: true,
+              },
+              {
+                userId: 102,
+                isFixed: false,
+              },
+            ],
+            schedulingType: SchedulingType.ROUND_ROBIN,
+          },
+        ],
+        organizer: fixedHost,
+        usersApartFromOrganizer: [roundRobinHost],
+        bookings: [
+          {
+            userId: 101, // Make fixed host busy
+            eventTypeId: 1,
+            startTime: `${getDate({ dateIncrement: 1 }).dateString}T04:00:00.000Z`,
+            endTime: `${getDate({ dateIncrement: 1 }).dateString}T04:45:00.000Z`,
+            status: "ACCEPTED",
+            attendees: [
+              {
+                email: "existing-booker@example.com",
+                name: "Existing Booker",
+              },
+            ],
+          },
+          {
+            userId: 102, // Make round robin host busy
+            eventTypeId: 1,
+            startTime: `${getDate({ dateIncrement: 1 }).dateString}T04:00:00.000Z`,
+            endTime: `${getDate({ dateIncrement: 1 }).dateString}T04:45:00.000Z`,
+            status: "ACCEPTED",
+            attendees: [
+              {
+                email: "existing-booker2@example.com",
+                name: "Existing Booker 2",
+              },
+            ],
+          },
+        ],
+      });
+
+      await createBookingScenario(scenarioData);
+
+      mockCalendar("googlecalendar", {
+        create: {
+          id: "MOCKED_GOOGLE_CALENDAR_EVENT_ID",
+          iCalUID: "MOCKED_GOOGLE_CALENDAR_ICS_ID",
+        },
+        busySlots: [],
+      });
+
+      const mockBookingData = getMockRequestDataForBooking({
+        data: {
+          eventTypeId: 1,
+          start: `${getDate({ dateIncrement: 1 }).dateString}T04:00:00.000Z`,
+          end: `${getDate({ dateIncrement: 1 }).dateString}T04:45:00.000Z`,
+          responses: {
+            email: booker.email,
+            name: booker.name,
+            location: { optionValue: "", value: "integrations:daily" },
+          },
+        },
+      });
+
+      await expect(
+        handleNewBooking({
+          bookingData: mockBookingData,
+        })
+      ).rejects.toThrow(ErrorCode.NoAvailableUsersFound);
+    },
+    timeout
+  );
+
+  test(
+    "should throw HostsUnavailableForBooking when Round Robin event has fixed hosts but no round robin host is available",
```

Comment:
We should have a more meaningful name here. 

`HostsUnavailableForBooking` and `NoAvailableUsersFound` mean the same thing and could be confusing in the future